test/padCharsEnd.js

```diff
@@ -0,0 +1,185 @@
+import * as RA from '../src';
+import {
+  padCharsEndStrInvoker,
+  padCharsEndStrPolyfill,
+} from '../src/padCharsEnd';
+import eq from './shared/eq';
+
+describe('padCharsEnd', function() {
+  specify('should pad string with given padString', function() {
+    eq(RA.padCharsEnd('-', 3, 'a'), 'a--');
+  });
+
+  specify('should be curried', function() {
+    eq(RA.padCharsEnd('*', 5, 'abc'), 'abc**');
+    eq(RA.padCharsEnd('*', 5)('abc'), 'abc**');
+    eq(RA.padCharsEnd('*')(5, 'abc'), 'abc**');
+    eq(RA.padCharsEnd('*')(5)('abc'), 'abc**');
+  });
+
+  context('given targetLength as 0', function() {
+    specify('should return original string', function() {
+      eq(RA.padCharsEnd('$', 0, 'abc'), 'abc');
+    });
+  });
+
+  context('given targetLength less than string length', function() {
+    specify('should return original string', function() {
+      eq(RA.padCharsEnd('$', 1, 'abc'), 'abc');
+    });
+  });
+
+  context('given targetLength less than equal to padString length', function() {
+    specify('should trim padString', function() {
+      eq(RA.padCharsEnd('123456', 6, 'abc'), 'abc123');
+      eq(RA.padCharsEnd('123456', 5, 'abc'), 'abc12');
+    });
+  });
+
+  context('given targetLength supplied as float', function() {
+    specify('should convert targetLength to integer', function() {
+      eq(RA.padCharsEnd('$', 7, 'abc'), 'abc$$$$');
+    });
+  });
+
+  context('given targetLength is not a number', function() {
+    specify('should return padded string', function() {
+      eq(RA.padCharsEnd('*', '5', 'abc'), 'abc**');
```

Comment:
This look more like `given targetLength is a value coercible to valid integer number`